zk/stages/stage_sequence_execute_blocks.go

```diff
@@ -136,6 +137,7 @@ func finaliseBlock(
 	}
 
 	txInfos := []blockinfo.ExecutedTxInfo{}
+	var txHash2SenderCache sync.Map
```

Comment:
Why we do use sync.Map rather than regular map. As far as I know we don't have any concurrency here?

In addition to that we can just use an array because the transactions are ordered (worth double checking this)

Reply:
yes, we should use regular map here.

Reply:
What about using an array? Using a map opens a question "What could happen if transaction count changes during execution"

Reply:
If using array, when transaction count changes during execution, the idx would be wrong. Using a map will not has this problem.

Reply:
If transaction count changes during execution we will have a HUGE problem. I've checked and it seems that it does not change but it is worth double checking. Only some fields in the transaction object changes but these fields. @revitteth Could you also check whether transaction count changes or not.

Reply:
Regardless of whether the number of transactions changes, it will not affect the logic here. We are simply trying to cache the already encountered transaction senders to reduce the number of times we calculate the sender. For transactions that haven't appeared, we will still continue to calculate their sender.

Reply:
I think this should work well - cache sender: if not present calculate it, otherwise look it up - unless I'm missing a hidden risk.